simapp/app_v2.go

```diff
@@ -245,9 +249,23 @@ func NewSimApp(
 
 	app.App = appBuilder.Build(logger, db, traceStore, baseAppOptions...)
 
-	if err := app.App.BaseApp.SetStreamingService(appOpts, app.appCodec, app.kvStoreKeys()); err != nil {
-		logger.Error("failed to load state streaming", "err", err)
-		os.Exit(1)
+	// register streaming services
+	streamingCfg := cast.ToStringMap(appOpts.Get(baseapp.StreamingTomlKey))
+	for service := range streamingCfg {
+		pluginKey := fmt.Sprintf("%s.%s.%s", baseapp.StreamingTomlKey, service, baseapp.StreamingABCIPluginTomlKey)
+		pluginName := strings.TrimSpace(cast.ToString(appOpts.Get(pluginKey)))
+		if len(pluginName) > 0 {
+			logLevel := cast.ToString(appOpts.Get(flags.FlagLogLevel))
+			plugin, err := streaming.NewStreamingPlugin(pluginName, logLevel)
+			if err != nil {
+				fmt.Printf("failed to load streaming plugin: %s", err)
+				os.Exit(1)
+			}
+			if err := baseapp.RegisterStreamingPlugin(app.BaseApp, appOpts, app.kvStoreKeys(), plugin); err != nil {
+				fmt.Printf("failed to register streaming plugin: %s", err)
+				os.Exit(1)
+			}
+		}
```

Comment:
can we pull this out into a util function in types or a different location. We should avoid adding more boiler plate to the app.go's

Reply:
moved this in `baseapp/streaming.go`